stages/org.osbuild.rpm

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python3
+
+import concurrent.futures
+import contextlib
+import json
+import os
+import pathlib
+import subprocess
+import sys
+
+
+RPM_CACHE_DIR = "/var/cache/org.osbuild.rpm"
+
+
+def download_package(pkg):
+    # some mirrors are broken sometimes. retry manually, because curl doesn't on 404
+    for _ in range(3):
+        curl = subprocess.run([
+            "curl",
+            "--silent",
+            "--show-error",
+            "--fail",
+            "--location",
+            "--remote-name",
+            "--write-out", "%{filename_effective}",
+            pkg["url"]
+        ], encoding="utf-8", cwd=RPM_CACHE_DIR, stdout=subprocess.PIPE, check=False)
+
+        if curl.returncode == 0:
+            filename = curl.stdout.strip()
+            break
+    else:
+        raise RuntimeError("Error downloading " + pkg["url"])
+
+    algorithm, checksum = pkg["checksum"].split(":", 1)
```

Comment:
Could we whitelist the algorithms we support?

Reply:
Good point: https://github.com/osbuild/osbuild/pull/142